### ISSUE TITLE
Update to use Mocha v.3

### ISSUE THREAD

**Issue by paul-barry-kenzan** [kenzanlabs/pipeline-test-node#27]
Version 3 is in Beta and will be available shortly. This pipeline should be updated to keep it's internal workings updated.

**Comment by obuckley-kenzan** [comment]
Very cool and good to know!

**Comment by obuckley-kenzan** [comment]
On a somewhat related note (?), what are your thoughts on #19 ?  If we can "require" consumers to be on NPM3, consumers can leverage the flat directory structure and not have to define their own dependencies on chai and sinon, for example.

I don't see any harm or backwards compatibility issues.  (eg. if on NPM2, you would already have these dependencies defined.  if on NPM3, you don't have to install them.  opt-in, if you will.)

**Comment by paul-barry-kenzan** [comment]
@obuckley-kenzan 
I'd much prefer that the consumer do as little setup as possible when using this pipeline "out-of-the-box". Forgive me if this is obvious. This npm version stuff always throws me for a loop, but is there a way to configure our npm `package.json` to always install what's needed regardless of npm version? Ideally, we want our package to install Mocha, Chai, Sinon, and any gulp helpers needed.

**Comment by obuckley-kenzan** [comment]
Not really.  

So in NPM <= 2, dependencies were installed in a nested fashion.  So any dependencies of this package would be installed in a _node_modules_ directory under it in the applications root _node_modules_ folder.

In NPM 3, all dependencies are installed in a flat directory structure (think of Bower, and Bower resolutions when two Bower components both install different versions of jQuery, for example.  In Bower, you would be forced to "resolve" to a specific version of jQuery and save that in bower.json)

So basically, with NPM how dependencies are installed and organized is based on the version the user has.  Different versions == different structure of node_modules.  I think this is why they had `peerDependencies` in NPM2, to ensure that the consumer installed related packages.  In NPM3, there are no more `peerDependencies`.

Hopefully that made it a little clearer.  :)

**Comment by paul-barry-kenzan** [comment]
I always knew the differences in directory structure (hard to miss when you open `node_modules` and see hundreds of directories). I guess my confusion always came with dependencies vs. peerDependencies and how each version of npm handle it. 

Your description helps clarify, for sure. 

So, my question is now, if we create a package.json with straight dependencies on what we need, no one should have an issue installing, right?

**Comment by obuckley-kenzan** [comment]
If they are on NPM3, yes.  

That said, I actually don't think there's anything to do in #19 .  We already have things setup as dependencies
https://github.com/kenzanlabs/pipeline-test-node/blob/master/package.json#L42

Maybe just a comment in the README encouraging consumers to be on NPM 3.x to avoid extra dependency management.

**Comment by obuckley-kenzan** [comment]
@bsawyer 
Something to review along with #31 ?

**Comment by bsawyer** [comment]
@obuckley-kenzan Im confused. This issue is for upgrading to Mocha 3.x correct? The NPM3 stuff sounds like it relates to #19's conversation.

**Comment by obuckley-kenzan** [comment]
@bsawyer 
Yes, maybe a little conflation I guess.  Was just thinking about what impact there would be to "shipping" our desired / preferred tools as `dependencies` leveraging NPM3 flat dependency structure.  Mocha 3 and NPM3 would be both breaking changes as far as this tool is considered.
